CAIPs/caip-358.md

```diff
@@ -39,135 +36,208 @@ The method transmits all the acceptable payment requests so the wallet can pick
 #### Request
 
 ```typescript
-type Hex = `0x${string}`;
-
+// Accepted Payment Options
 type PaymentOption = {
-  asset: string; 
-  amount: Hex;
+  asset: string;
+  amount: string;
   recipient: string;
-}
+  types: string[];
+};
 
-// JSON-RPC Request
-type PayRequest = {
-  version: string;
-  orderId?: string;
-  acceptedPayments: PaymentOption[]; 
+// JSON-RPC Request Params
+type RequestParams = {
+  version: integer;
+  orderId: string;
   expiry: number;
-}
-
+  paymentOptions: PaymentOption[];
+};
 ```
 
-The application **MUST** include:
+The following request parameters are defined for `version=1` as:
 
-- At least one entry in the `acceptedPayments` array
-- `expiry` timestamp for the payment request
+- `version` - this field is an integer and **MUST** be present to define which of the following parameters are optional or required.
+- `orderId` - this field **MUST** uniquely identify an order which this payment request is linked to and **MUST NOT** be longer than 128 characters. It does not require global uniqueness, but it's **RECOMMENDED** to use a UUIDv4 if global uniqueness is necessary.
+- `expiry` - this field **MUST** be a UNIX timestamp (in seconds) after which the payment request is considered expired. It **SHOULD** use an expiry of at least 5 minutes (300 seconds).
+- `paymentOptions` - this field **MUST** be an array of `PaymentOption` objects with at least one entry. Each element in the array represents a payment option the wallet may choose to complete the payment, each with independent parameters.
 
-The application **MAY** include:
+**Note:** Merchants **CAN** signal `PaymentOption` preference by ordering the paymentOptions array. JSON-RPC guarantees array order, but wallets are **NOT REQUIRED** to honor it. Wallets **MAY** offer swaps or bridges to help users match the merchant’s preferred option.
 
-- An `orderId` that uniquely identifies this payment request. If provided, `orderId` **MUST NOT** be longer than 128 characters.
+For `PaymentOption` parameters these are defined for `version=1` as:
 
- When `orderId` is provided, it **MUST** be a string and implementations **SHOULD** ensure this ID is unique across their system to prevent collisions.
+- `asset` - this field **MUST** follow the assetId [CAIP-19][] spec which also includes the [CAIP-2][] chainId prefix.
+- `amount` - this field **MUST** be a Hex string representing the amount in the smallest denomination of its asset.
+- `recipient` - this field **MUST** be a chain-specific address present in the chain referred in the `asset` field.
+- `types` - this field **MUST** be an array of strings defining different transfer authorization types.
 
-The `acceptedPayments` field **MUST** be an array of `PaymentOption` objects. Each element in the array represents a payment option that the wallet can choose from to complete the payment.
+The exclusive list of Transfer Types supported in `version=1` are the following:
```

Comment:
We could make this non-exclusive and extensible easily with a defined "private types" approach. Right now we're effectively locking this to the types of today rather than the types of what's possible.

A good example of what this might look like is: https://datatracker.ietf.org/doc/html/rfc7519#section-4.3

Reply:
Honestly I would say that is the best long-term solution but historically in crypto we have seen this go very bad very quickly

The lack of governance where these "identifiers" can cause a lot of issues and this is very apparent on Ethereum with capabilities which is a melting pot of functionality that barely qualifies as a capability

In this scenario we are locking a narrow scope of accepted payment types that cover 95% of use-cases if not more

That being said this PR is still in Draft status so we can still make a follow-up PR to attempt a "private types" approach